### PR TITLE
docs: polish up guide docs

### PR DESCRIPTION
**Pull request by fi3ework**
## Summary

Changes worth mentioning:
Remove https://www.rspack.dev/guide/tech/nodejs and move the use method of [node-polyfill-webpack-plugin](https://www.npmjs.com/package/node-polyfill-webpack-plugin) to `resolve.fallback`.

## Checklist

- [ ] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

**Review by SyMind — DISMISSED** [review]

File: website/docs/en/guide/tech/solid.mdx [hunk 1]
```diff
@@ -2,14 +2,14 @@
 
 ## How to Use
 
-Rspack provides two solutions to support Solid:
+Rspack provides two solutions to support SolidJS:
```
Comment (chenjiahan) [inline website/docs/en/guide/tech/solid.mdx:5]: We can just call `SolidJS` as `Solid`:

image

https://www.solidjs.com/guides/getting-started

File: website/docs/zh/config/resolve.mdx [hunk 2]
```diff
@@ -120,7 +120,67 @@ module.exports = {
...
+  resolve: {
+    fallback: {
+      abc: false, // 不为 abc 引入 polyfill
+      xyz: path.resolve(__dirname, 'path/to/file.js'), // 为 xyz 引入 polyfill
+    },
+  },
+};
+```
+
+Rspack 默认不会为 Node.js 核心模块提供 polyfills，这意味着如果你在浏览器或类似环境中运行的代码中使用它们，你将不得不从 NPM 安装兼容的模块并自行包含它们。
```
Comment (chenjiahan) [inline website/docs/zh/config/resolve.mdx:137]:
```suggestion
Rspack 默认不会为 Node.js 核心模块提供 polyfills，这意味着如果你在浏览器或类似环境中运行的代码中使用它们，你将需要从 NPM 安装兼容的模块并自行包含它们。
```

File: website/docs/zh/guide/tech/json.mdx [hunk 3]
```diff
@@ -10,5 +12,20 @@ JSON 是 Rspack 的一等公民，你可以直接导入，例如：
 
 ```ts title="index.js"
 import json from './example.json';
-json.foo; // bar
+console.log(json.foo); // "bar"
+```
+
+## 具名引用
+
+在非 `.mjs` 的非严格 ESM 文件中，你可以直接导入 JSON 中的属性
```
Comment (chenjiahan) [inline website/docs/zh/guide/tech/json.mdx:20]:
```suggestion
在非 `.mjs` 的非严格 ESM 文件中，你可以直接导入 JSON 中的属性。
```

File: website/docs/zh/guide/tech/json.mdx [hunk 4]
```diff
@@ -10,5 +12,20 @@ JSON 是 Rspack 的一等公民，你可以直接导入，例如：
 
 ```ts title="index.js"
 import json from './example.json';
-json.foo; // bar
+console.log(json.foo); // "bar"
+```
+
+## 具名引用
```
Comment (chenjiahan) [inline website/docs/zh/guide/tech/json.mdx:18]:
```suggestion
## 具名导入
```

Comment (fi3ework) [reply website/docs/zh/guide/tech/json.mdx:18]: Whether you believe or not, I purposely aligned it with Rsbuild 🤣. https://rsbuild.dev/zh/guide/basic/json-files#%E5%85%B7%E5%90%8D%E5%BC%95%E7%94%A8

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
👍

**Pull request merged**